### PR TITLE
[Switch] Fix OS_Switch::show_virtual_keyboard

### PR DESCRIPTION
**Pull request by hokaze**
Implementation of OS_Switch::show_virtual_keyboard is wrong, arguments list doesn't match the format os.show_virtual_keyboard expects:

os.h
`virtual void show_virtual_keyboard(const String &p_existing_text, const Rect2 &p_screen_rect = Rect2(), bool p_multiline = false, int p_max_input_length = -1, int p_cursor_start = -1, int p_cursor_end = -1);`

os_switch.h:
`virtual void show_virtual_keyboard(const String &p_existing_text, const Rect2 &p_screen_rect = Rect2(), int p_max_input_length = -1);`

This is why the show_virtual_keyboard method never executes any of the code within, despite everything else about the inline software keyboard from libnx otherwise largely working.

### CODE REVIEW

File: platform/switch/os_switch.cpp [hunk 1]
```diff
@@ -456,11 +456,10 @@ void OS_Switch::run() {
 			}
 		}
 
+        swkbdInlineUpdate(&inline_keyboard, NULL);
```
Comment (halotroop2288) [inline platform/switch/os_switch.cpp]: Why did you move this up?

File: platform/switch/os_switch.cpp [hunk 2]
```diff
@@ -478,15 +477,15 @@ bool OS_Switch::has_virtual_keyboard() const {
...
 		return 0;
 	}
-	return 300;
+	return 400;
 }
 
-void OS_Switch::show_virtual_keyboard(const String &p_existing_text, const Rect2 &p_screen_rect, int p_max_input_length) {
-	if (!g_swkbd_open) {
+void OS_Switch::show_virtual_keyboard(const String &p_existing_text, const Rect2 &p_screen_rect, bool p_multiline, int p_max_input_length, int p_cursor_start, int p_cursor_end) {
+	printf("Showing kbd!\n");
```
Comment (halotroop2288) [inline platform/switch/os_switch.cpp]: This doesn't need a debug print, now that it's fixed.

**Review by halotroop2288 — CHANGES_REQUESTED** [review]

File: platform/switch/os_switch.cpp (hunk 1 above)
Comment (hokaze) [reply platform/switch/os_switch.cpp]: Ah, my bad, that was from when I first tested Stary's suggestion from a few months ago of trying to update the keyboard before the input buffer was flushed

As that was originally theorised to be the problem before you spotted the actual issue - I forgot to revert that, ditto for one of my own printf lines

File: platform/switch/os_switch.cpp (hunk 2 above)
Comment (hokaze) [reply platform/switch/os_switch.cpp]: Agreed, will fix.

File: platform/switch/os_switch.cpp (hunk 1 above)
Comment (halotroop2288) [reply platform/switch/os_switch.cpp]: Also note the failed style check when you revert this. It needs to be indented with \t not four spaces.

Comment (halotroop2288) [reply platform/switch/os_switch.cpp]: You kinda forgot to fix the rest of the indentation but I got it.

**Comment by hokaze** [comment]
Thanks for doing the cleanup, I somehow completely missed that I hadn't actually removed the printf or the other lines with wrong indents.

I'm taking this as a learning experience for next time, namely to keep in mind the style conventions and not just autopilot there, to review my changes more thoroughly before pushing and to probably do any further coding when more rested and not rocking 2 nights insomnia in a row 😅

**Review by halotroop2288 — APPROVED** [review]
